sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/models/WebApplicationFirewallPolicy.java

```diff
@@ -0,0 +1,533 @@
+import com.azure.resourcemanager.resources.fluentcore.model.Refreshable;
+import com.azure.resourcemanager.resources.fluentcore.model.Updatable;
+import reactor.core.publisher.Flux;
+
+import java.util.List;
+
+/** Entry point for Web Application Firewall Policy. */
+public interface WebApplicationFirewallPolicy
+    extends GroupableResource<NetworkManager, WebApplicationFirewallPolicyInner>,
+    Updatable<WebApplicationFirewallPolicy.Update>,
+    Refreshable<WebApplicationFirewallPolicy> {
+
+    /** @return mode of the Web Application Firewall Policy */
+    WebApplicationFirewallMode mode();
+
+    /** @return whether request body inspection is enabled */
+    boolean isRequestBodyInspectionEnabled();
+
+    /**
+     * Max request body size limit in KB.
+     *
+     * @return request body size limit in KB */
+    Integer requestBodySizeLimitInKb();
+
+    /**
+     * Max file upload size limit, in MB.
+     *
+     * @return file upload limit in MB
+     */
+    Integer fileUploadSizeLimitInMb();
+
+    /** @return Web Application Firewall Policy settings */
+    PolicySettings getPolicySettings();
+
+    /** @return Web Application Firewall managed rules */
+    ManagedRulesDefinition getManagedRules();
+
+    /** @return whether this policy is enabled */
+    boolean isEnabled();
+
+    /** @return an immutable list of application gateway resource ids associated with this Web Application Firewall Policy */
+    List<String> getAssociatedApplicationGatewayIds();
+
+    /**
+     * Get an immutable list of application gateways associated with this Web Application Firewall Policy by calling REST API.
+     *
+     * @return an immutable list of application gateways associated with this Web Application Firewall Policy
+     */
+    List<ApplicationGateway> getAssociatedApplicationGateways();
+
+    /**
+     * Get a list of application gateways associated with this Web Application Firewall Policy by calling REST API
+     * in async manner.
+     *
+     * @return {@link Flux} of application gateways associated with this Web Application Firewall Policy
+     */
+    Flux<ApplicationGateway> getAssociatedApplicationGatewaysAsync();
+
+    /**
+     * The entirety of the Web Application Firewall Policy definition.
+     */
+    interface Definition
+        extends DefinitionStages.Blank,
+        DefinitionStages.WithGroup,
+        DefinitionStages.WithCreate,
+        DefinitionStages.WithRequestBodyOrCreate,
+        DefinitionStages.WithManagedRulesOrCreate {
+    }
+
+    /** Grouping of Web Application Gateway stages. */
+    interface DefinitionStages {
+        /** The first stage of an application gateway definition. */
+        interface Blank extends DefinitionWithRegion<WithGroup> {
+        }
+
+        /** The stage of a Web Application Firewall Policy definition allowing to specify the resource group. */
+        interface WithGroup extends GroupableResource.DefinitionStages.WithGroup<WithManagedRules>{
```

Comment:
Is the flow, that user will need to at least have one rule in `WithManagedRules`, then they can optionally do a bot rule `WithBotProtection`?
And underlying, both be under `managedRules`?

Can user set bot rule under `WithManagedRules` flow?

Reply:
Yes, they can. `WithBotProtection` acts like a shortcut, kind of like Portal's when creating application gateway along with the waf policy:
<img width="835" alt="Screenshot 2023-06-20 at 11 08 00 AM" src="https://github.com/Azure/azure-sdk-for-java/assets/92354331/5f15f6c8-746d-4bee-814a-ebbadb89650b">

It's before the introduction of `WithManagedRules` though. 
Since we already have it, suppose we can get rid of it, put it into `KnownWebApplicationFirewallManagedRuleSet`, and add an additional `enableBotProtection` to `withNewWebApplicationFirewallPolicy` when creating `ApplicationGateway`?

Reply:
I assume not `KnownWebApplicationFirewallManagedRuleSet`? It is just one managed rule. While your `WithManagedRules` can define multiple rule sets.

Current place seems fine?